hack/enable-kibana-msearch-access.sh

```diff
@@ -0,0 +1,192 @@
+    set -x
+    curl_output() {
+        python -mjson.tool
+    }
+else
+    curl_output() {
+        cat > /dev/null 2>&1
+    }
+fi
+
+# $1 - es pod name
+# $2 - es endpoint
+# rest - any args to pass to curl
+function curl_es() {
+    local pod="$1"
+    local endpoint="$2"
+    shift; shift
+    local args=( "${@:-}" )
+
+    local secret_dir="/etc/elasticsearch/secret/"
+    oc exec "${pod}" -- curl --silent --insecure "${args[@]}" \
+                             --key "${secret_dir}admin-key"   \
+                             --cert "${secret_dir}admin-cert" \
+                             "https://localhost:9200${endpoint}"
+}
+
+# $1 - es pod name
+# $2 - es endpoint
+# rest - any args to pass to curl
+function curl_es_input() {
+    local pod="$1"
+    local endpoint="$2"
+    shift; shift
+    local args=( "${@:-}" )
+
+    local secret_dir="/etc/elasticsearch/secret/"
+    oc exec -i "${pod}" -- curl --silent --insecure "${args[@]}" \
+                                --key "${secret_dir}admin-key"   \
+                                --cert "${secret_dir}admin-cert" \
+                                "https://localhost:9200${endpoint}"
+}
+
+usage() {
+    echo Usage: $0 user-name [project-1 ... project-N]
+    echo If no projects are specified, the msearch access will be removed
+}
+
+if [ -z "${1:-}" ] ; then
+    usage
+    exit 1
+fi
+
+if oc get users "$1" > /dev/null 2>&1 ; then
+    echo Using user "$1" . . .
+else
+    echo Error: user "$1" not found
+    usage
+    exit 1
+fi
+
+user="$1" ; shift
+
+for proj in "$@" ; do
+    if oc get project "$proj"  > /dev/null 2>&1 ; then
+        echo Using project "$proj" . . .
+    else
+        echo Error: project "$proj" not found
+        usage
+        exit 1
+    fi
+done
+
+espod=$( oc get pods --selector component=es -o jsonpath='{ .items[0].metadata.name }' )
+
+if [ -z "$espod" ] ; then
+    echo Error: no Elasticsearch pod found
+    usage
+    exit 1
+fi
+
+role_name="multi_index_access_for_user_$user"
+
+config_index_name=$( oc exec $espod -- python -c "import yaml; print yaml.load(open('/usr/share/elasticsearch/config/elasticsearch.yml'))['searchguard']['config_index_name']" )
+if [ -z "$config_index_name" ] ; then
+    echo Error: could not extract the searchguard index name from $espod /usr/share/elasticsearch/config/elasticsearch.yml
+    usage
+    exit 1
+fi
+
+sg_index=$( oc exec $espod -- bash -c "eval 'echo $config_index_name'" )
+if [ -z "$sg_index" ] ; then
+    echo Error: could not convert "$config_index_name" in $espod to the searchguard index name
+    usage
+    exit 1
+fi
+
+if [ "$#" -gt 0 ] ; then
+    uuids=$( oc get project "$@" -o jsonpath='{.items[*].metadata.uid}' )
+    if [ -z "$uuids" ] ; then
+        echo Error: could not get the project uuids for the given projects
+        usage
+        exit 1
+    fi
+fi
+
+echo Input looks good - fixing ACLs . . .
+
+curl_es $espod /$sg_index/roles/0 | PROJECTS="$@" UUIDS="${uuids:-}" python -c '
```

Comment:
This exhibits great bash fu, but is there a reason why having separate python files to contain these programs is not an option?